query-node/mappings/src/content/utils.ts

```diff
@@ -166,6 +173,89 @@ async function processVideoSubtitleAssets(
   }
 }
 
+async function checkAppActionNonce<T>(
+  ctx: EventContext & StoreContext,
+  entity: T,
+  actionOwnerId: string | undefined,
+  appAction: DecodedMetadataObject<IAppAction>
+): Promise<boolean> {
+  if (!appAction.metadata?.nonce || !actionOwnerId) {
+    return false
+  }
+
+  if (appAction.contentMetadata?.videoMetadata) {
+    return (
+      (await ctx.store.getMany(Video, { where: { channel: { ownerMember: { id: actionOwnerId } } } })).length ===
```

Comment:
I wonder how will it imply for curator group ownership. If we add the fields to the member, then we shouldn't validate curator group none at all?

Maybe instead of fetching video/channel as it can be deleted we would fetch `..._CreatedEvent` and just validate against its length.

Reply:
1. The reason I didn't mention curator groups is that I noticed they are not fully supported currently, for example, in
https://github.com/Joystream/joystream/pull/4631/files#diff-526bb3b7420aeed56e6c5bfa94e07ce3a8302d6ed7ad874b394b2185d542a003R102 you are only passing `ownerMember?.id` and in https://github.com/Joystream/joystream/pull/4631/files#diff-1b8138e018fcb5011c92af46bc3ed5a7a04579644c72675900d69dbdfe6b5fa3R182 you make the nonce check fail in case there is no owner id provided, which makes it impossible to pass this check in case of curator-group-owned channels anyway. If you wish to add support for curator-group-owned channels you need a similar counter in `CuratorGroup` entity perhaps.

2. Counting `ChannelCreated` / `VideoCreated` events would be another way to go about it, but we currently don't store those in the processor database, so I think that's a more complex change.